ios/Classes/PosthogFlutterPlugin.swift

```diff
@@ -0,0 +1,286 @@
+import Flutter
+import UIKit
+import PostHog
+
+public class PosthogFlutterPlugin: NSObject, FlutterPlugin {
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let channel = FlutterMethodChannel(name: "posthog_flutter", binaryMessenger: registrar.messenger())
+    let instance = PosthogFlutterPlugin()
+    initPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+    
+   public static func initPlugin(){
+    // Initialise Poshog
+       let postHogApiKey = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.API_KEY") as? String ?? ""
+       let postHogHost = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.POSTHOG_HOST") as? String ?? ""
+         let postHogCaptureLifecyleEvents = Bundle.main.object(forInfoDictionaryKey: "com.posthog.posthog.TRACK_APPLICATION_LIFECYCLE_EVENTS") as? Bool ?? false
+         
+         let config = PostHogConfig(
+           apiKey: postHogApiKey,
+           host: postHogHost
+         )
+         config.captureApplicationLifecycleEvents = postHogCaptureLifecyleEvents
+         PostHogSDK.shared.setup(config)
+       //
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    switch call.method {
+    case "getPlatformVersion":
+        getPlatformVersion(call, result: result)
+    case "getFeatureFlag":
+        getFeatureFlag(call, result: result)
+    case "isFeatureEnabled":
+        isFeatureEnabled(call, result: result)
+    case "getFeatureFlagPayload":
+        getFeatureFlagPayload(call, result: result)
+    case "getFeatureFlagAndPayload":
+        getFeatureFlagAndPayload(call, result: result)
+    case "identify":
+        identify(call, result: result)
+    case "capture":
+        capture(call, result: result)
+    case "screen":
+        screen(call, result: result)
+    case "alias":
+        alias(call, result: result)
+    case "distinctId":
+        distinctId(call, result: result)
+    case "reset":
+        reset(call, result: result)
+    case "enable":
+        enable(call, result: result)
+    case "disable":
+        disable(call, result: result)
+    case "reloadFeatureFlags":
+        reloadFeatureFlags(call, result: result)
+    case "group":
+        group(call, result: result)
+    case "register":
+        register(call, result: result)
+    default:
+      result(FlutterMethodNotImplemented)
+    }
+  }
+
+    private func getPlatformVersion(
+        _ call: FlutterMethodCall,
+        result: @escaping FlutterResult
+    ){
+        result("iOS " + UIDevice.current.systemVersion)
+    }
+    
+    private func getFeatureFlag(
+        _ call: FlutterMethodCall,
+        result: @escaping FlutterResult
+    ){
+            
+        if let args = call.arguments as? Dictionary<String, Any>,
```

Comment:
```suggestion
        if let args = call.arguments as? [String: Any],
```

This cast is shorter and preferable.